pkg/apply/apply.go

```diff
@@ -3,19 +3,83 @@ package apply
 import (
 	"context"
 	"fmt"
+	corev1 "k8s.io/api/core/v1"
 	"log"
 
 	"github.com/pkg/errors"
 
+	"gopkg.in/yaml.v2"
 	"k8s.io/apimachinery/pkg/api/equality"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/types"
 	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// Find existing object or create if if it doesn't exists
-func findOrCreateObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstructured) (*uns.Unstructured, string, error) {
+type ObjectType int
+
+const (
+	ADDRESSSPOOL ObjectType = iota
+	BGPPEER
+)
+
+func UpdateConfigMapObjs(ctx context.Context, client k8sclient.Client, objs []*uns.Unstructured, objtype ObjectType, namespace string) error {
+	existingStruct, updateStruct := ConfigMapData{}, ConfigMapData{}
```

Comment:
I think the "Struct" suffix is unnecessary, can use (cur,upd) or (current,updated) I guess